### ISSUE TITLE
Strains not displayed in multi-allele genotypes

### ISSUE THREAD

**Issue by jseager7** [pombase/canto#1700]
**Summary:** the strain name is not passed into the function that stores a multi-allele genotype. See #1703.

After combining two single allele genotypes (with the same strain) into a multi-allele genotype, the resulting row in the multi-allele genotype list doesn't show anything in the 'Strain' column: it shows a single empty cell that does not span the expected number of rows:

![multi_allele_strain_bug](https://user-images.githubusercontent.com/37659591/49151257-59b2a280-f307-11e8-9e31-8e67d37be0cc.PNG)

**Comment by jseager7** [comment]
I've fixed the row spanning problem, but I'm stuck on trying to get the strain to display. The strain is currently stored on the following line in `genotypeListRowCtrl`:

https://github.com/pombase/canto/blob/f161098a78ed069410c95ecb4816140479d8fe3f/root/static/js/canto-modules.js#L4585

Then the template references this in a table cell:

```html
  <td rowspan="{{genotype.alleles.length}}">{{ strain }}</td>
```

This works fine for single allele genotypes, but when the controller tries to render a multi-allele genotype, `strain` is `null` for some reason. Maybe the code neglects to set the strain for multi-allele genotypes. Here's a JSON dump of the offending multi-allele genotype:

```json
{
  "genotype_id": 17,
  "display_name": "Alt a1delta endoPGdelta",
  "metagenotype_count": 0,
  "allele_string": "Alt a1delta endoPGdelta",
  "alleles": [
    {
      "description": null,
      "expression": null,
      "name": "Alt a1delta",
      "allele_id": 6,
      "display_name": "Alt a1delta",
      "type": "deletion",
      "gene_id": 6,
      "gene_display_name": "Alt a1",
      "uniquename": "A0A1S5QMS8:5b38aa5238216ac8-1"
    },
    {
      "gene_id": 7,
      "gene_display_name": "endoPG",
      "uniquename": "A0A1S5QMS6:5b38aa5238216ac8-1",
      "allele_id": 7,
      "display_name": "endoPGdelta",
      "type": "deletion",
      "name": "endoPGdelta",
      "description": null,
      "expression": null
    }
  ],
  "background": null,
  "name": null,
  "organism": {
    "full_name": "Alternaria alternata",
    "scientific_name": "Alternaria alternata",
    "genes": [
      {
        "primary_identifier": "A0A1S5QMS8",
        "gene_id": 6,
        "display_name": "Alt a1",
        "primary_name": "Alt a1"
      },
      {
        "gene_id": 7,
        "primary_identifier": "A0A1S5QMS6",
        "primary_name": "endoPG",
        "display_name": "endoPG"
      }
    ],
    "common_name": "Alternaria rot fungus",
    "pathogen_or_host": "pathogen",
    "taxonid": "5599"
  },
  "annotation_count": 0,
  "identifier": "5b38aa5238216ac8-genotype-17",
  "strain_name": null,
  "id_or_identifier": 17,
  "feature_id": 17,
  "$$hashKey": "object:92"
}
```

**Comment by jseager7** [comment]
I checked the _curs_ database for the curation session, and all of the multi-allele genotypes have NULL values for the strain field. Here's proof:

![no_multi_allele_strains](https://user-images.githubusercontent.com/37659591/49223803-baa9ab80-f3d7-11e8-8aec-6a02a218c98a.PNG)

After more investigation, I think I've found the offending code in the `storePromise` variable in `combineGenotypes`:

https://github.com/pombase/canto/blob/f161098a78ed069410c95ecb4816140479d8fe3f/root/static/js/canto-modules.js#L4689-L4690

If we compare this to the full list of expected arguments for `CursGenotypeList.storeGenotype`, we can see that some of them are missing:

https://github.com/pombase/canto/blob/f161098a78ed069410c95ecb4816140479d8fe3f/root/static/js/canto-modules.js#L382-L383

Here's a summary:

CursGenotypeList|combineGenotypes
|---|---|
toaster|toaster
$http|$http
genotype_id|undefined
genotype_name|undefined
genotype_background|newBackground
alleles|allelesForGenotype
taxonid|
strain_name|

So the `taxonid` and the `strain_name` aren't passed to `CursGenotypeList` in the first place. 

@kimrutherford can this be fixed just by passing in the strain of one of the genotypes in `$scope.genotypeList?` A lot of the existing code seems to be using `$.map` and `$.grep` to collect values from every genotype, but since we're planning to enforce the strain being equal for every allele in a multi-allele genotype, we shouldn't need to check them all.

However I've just realised we don't have an issue about enforcing this strain equality constraint, so I'll make sure to open one, now that I understand where the check should be.

**Comment by jseager7** [comment]
This is fixed by #1707. Multi-allele genotypes now display the strain of the organism:

![multi_strain_fix](https://user-images.githubusercontent.com/37659591/49227528-34926280-f3e1-11e8-8cdf-15586406e32c.PNG)
